scipy/stats/_distn_infrastructure.py

```diff
@@ -1230,87 +1225,6 @@ def moment(self, order=None, *args, **kwds):
-        #    users to rely on this error in their code, though, so I don't see
-        #    this as a backward compatibility break.
-        # 2. If only `n` is provided (as a keyword argument), this would have
-        #    been an error in the past because `n` would have been treated as
-        #    the order of the moment while the shape parameter would be
-        #    missing. It is still the same type of error, but for a different
-        #    reason: now, `n` is treated as the shape parameter while the
-        #    order of the moment is missing.
-        # 3. If only `order` is provided, no special treament is needed.
-        #    Clearly this value is intended to be the order of the moment,
-        #    and the rest of the function will determine whether `n` is
-        #    available as a shape parameter in `args`.
-        # 4. If neither `n` nor `order` is provided, this would have been an
-        #    error (order of the moment is not provided) and it is still an
-        #    error for the same reason.
-
-        # Case B: the distribution does _not_ accept `n` as a shape
-        # 1. If both `order` and `n` are provided, this was an error, and it
-        #    still is an error: two values for same parameter.
-        # 2. If only `n` is provided (as a keyword argument), this was OK and
-        #    is still OK, but there shold now be a `DeprecationWarning`. The
-        #    value of `n` should be removed from `kwds` and stored in `order`.
-        # 3. If only `order` is provided, there was no problem before providing
-        #    only the first argument of `moment`, and there is no problem with
-        #    that now.
-        # 4. If neither `n` nor `order` is provided, this would have been an
-        #    error (order of the moment is not provided), and it is still an
-        #    error for the same reason.
-        if not got_order and ((not got_keyword_n)  # A4 and B4
-                              or (got_keyword_n and has_shape_n)):  # A2
-            message = ("moment() missing 1 required "
-                       "positional argument: `order`")
-            raise TypeError(message)
-
-        if got_keyword_n and not has_shape_n:
-            if got_order:  # B1
-                # this will change to "moment got unexpected argument n"
-                message = "moment() got multiple values for first argument"
-                raise TypeError(message)
-            else:  # B2
-                message = ("Use of keyword argument 'n' for method 'moment is"
-                           " deprecated and will be removed in SciPy 1.11.0. "
-                           "Use first positional argument or keyword argument"
-                           " 'order' instead.")
-                order = kwds.pop("n")
-                warnings.warn(message, DeprecationWarning, stacklevel=2)
         n = order
         # No special treatment of A1, A3, or B3 is needed because the order
```

Comment:
This comment can be removed. (Comparing against https://github.com/scipy/scipy/pull/13490 is how I'm finding these.)